src/System.Memory/tests/System.Memory.Tests.csproj

```diff
@@ -167,6 +164,11 @@
       <Link>ReadOnlySpan\StringTests.cs</Link>
     </Compile>
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetGroup)' != 'netfx'">
+    <Compile Include="$(CommonTestPath)\Tests\System\StringTests.netcoreapp.cs">
```

Comment:
@ahsonkhan added netcoreapp only tests as above, now i tried to add also on System.Runtime but there is already a file named StringTests.netcoreapp.cs...i think is not correct merge with new tests(these are String/Span tests), what do you think?Rename file?

Reply:
I think this actually reduces the test coverage since the span API test (which is the primary focus here) won't run on netfx, even though they should.

Given that (and the file naming conflict), let's revert back to what you originally had (without the exception for netfx). The string.Contains method already has decent test coverage.

Reply:
i agree...string.Contains are old aged and tested, here the focus is on Span and semantic differences, IMO see that there are "differences"(missing overload) on api between netcoreapp and netfx could be better(more clear). Again run netfx on Common tests is not so easy, there are other issue involving other area. 
Having said that if you agree i'll revert and we'll have Contains tests String where "possible", we could use this "rule" from now. Do you agree?

Reply:
> Having said that if you agree i'll revert and we'll have Contains tests String where "possible", we could use this "rule" from now. Do you agree?

I agree. Even though seeing the semantic differences for various platforms could be useful, since it introduces test complexity we shouldn't try to solve it in this PR. When I initially proposed changing it, I thought it would be easier. Sorry for pushing us in that direction. Reverting back to something like this is fine by me:
https://github.com/dotnet/corefx/pull/29935/files/3dafdecc1a431cd012f598a444ab38977e7c1e62 + https://github.com/dotnet/corefx/pull/29935/commits/b8e1e09e2ae332d33622c60fba8366db23ee6d70 + some of the whitespace cleanup/etc.

Reply:
>Sorry for pushing us in that direction. 

No problem!

> Like you said, we have tests for contains already: ...

Maybe is not possible completely avoid duplication...or rather not today.

I try to work on this PR, i did a merge with master so revert is not so smootly... if it gets messy i'll close this and reopen a new PR, do you agree?

Reply:
Doing git revert would be messy. I would just add a new commit with the changes to remove the netfx definitions. We will squash the commits right before merging anyway so the commit history within the PR is not that important.

Reply:
> We will squash the commits right before merging anyway so the commit history within the PR is not that important.

In past i tried to squash with a "merge from master" commit and after that i saw more file modification than expected...hower we'll try again, maybe i made mistake.

Reply:
I meant, we use the "Squash and merge" option on GitHub when merging the PR. You don't have to explicitly squash the commits yourself.

![image](https://user-images.githubusercontent.com/6527137/40944942-82a2896c-680b-11e8-9c0c-9efd86b320f1.png)

Reply:
Ah ok!understood!